pgjdbc/src/test/java/org/postgresql/test/jdbc42/TimestampUtilsTest.java

```diff
@@ -62,6 +64,30 @@ public void testToLocalTime() throws SQLException {
     assertEquals(LocalTime.parse("23:59:59.999999999"), timestampUtils.toLocalTime("24:00:00"));
   }
 
+  @Test
+  public void testLocalDateTimeRounding() {
+    TimestampUtils timestampUtils = createTimestampUtils();
+
+    assertEquals(LocalDateTime.parse("2022-01-04T08:57:13.123457"), timestampUtils.round(LocalDateTime.parse("2022-01-04T08:57:13.123456500")));
+    assertEquals(LocalDateTime.parse("2022-01-04T08:57:13.123456"), timestampUtils.round(LocalDateTime.parse("2022-01-04T08:57:13.123456499")));
+    assertEquals(LocalDateTime.parse("2022-01-04T08:57:14"), timestampUtils.round(LocalDateTime.parse("2022-01-04T08:57:13.999999501")));
```

Comment:
It would help to extract `assertLocalDateTimeRounding(String inputTimestamp, String expectedOutputTimestamp)`, so the parsing logic is not duplicated, and the failure message can include the original input.

The way you write `assertEquals(Local...` assertions would yield hard-to-understand messages like "expected 2... got 2..."